js/index.js

```diff
@@ -1,39 +1,93 @@
 // ITERATION 1
 
+const { ConnectionEmittedEvents } = require("puppeteer");
+
 function updateSubtotal(product) {
   console.log('Calculating subtotal, yey!');
+  
+  const quantity = product.querySelector('.quantity input').innerHTML;
+  const price = product.querySelector('.price span').innerHTML;
+
+  const quantityValue = quantity.innerHTML;
+  const priceValue = price.innerHTML;
+
+  const calcSubtotal = quantityValue * priceValue;
+
+  const displaySubTotal = product.querySelector('.subtotal span');
+
+  displaySubTotal.innerHTML = calcSubtotal;
+
+  return calcSubtotal;
+  
 
-  //... your code goes here
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
+function calculateAll() {
+  
   // ITERATION 2
-  //... your code goes here
+  const productsTotality = document.getElementsByClassName('.product')
```

Comment:
Que tal Javier, buen esfuerzo, aquí podíamos hacer un ciclo for para poder ir acumulando los valores de cada producto:
`const product = document.getElementsByClassName('product')
  let total = 0
  for (const elemento of product) {
    total += updateSubtotal(elemento)
  }`